crates/router/src/core/payment_methods/validator.rs

```diff
@@ -62,18 +62,28 @@ pub async fn validate_request_and_initiate_payment_method_collect_link(
 
     // Fetch all configs
     let default_config = &state.conf.generic_link.payment_method_collect;
+
+    #[cfg(all(
+        any(feature = "v1", feature = "v2"),
+        not(feature = "merchant_account_v2")
+    ))]
     let merchant_config = merchant_account
         .pm_collect_link_config
         .as_ref()
         .map(|config| {
-            config
-                .clone()
-                .parse_value::<admin::BusinessCollectLinkConfig>("BusinessCollectLinkConfig")
+            common_utils::ext_traits::ValueExt::parse_value::<admin::BusinessCollectLinkConfig>(
+                config.clone(),
+                "BusinessCollectLinkConfig",
+            )
         })
         .transpose()
         .change_context(errors::ApiErrorResponse::InvalidDataValue {
             field_name: "pm_collect_link_config in merchant_account",
         })?;
+
+    #[cfg(all(feature = "v2", feature = "merchant_account_v2"))]
+    let merchant_config = Option::<admin::BusinessCollectLinkConfig>::None;
```

Comment:
If the pm_collect_link_config is deprecated, this whole function implementation itself should change right?

Reply:
This uses the default fallback. @kashif-m can you provide some more info here?